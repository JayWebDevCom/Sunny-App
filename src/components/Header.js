// import libraries for making a Component
import React from 'react';
import { Text, View } from 'react-native';

// make a coponent
const Header = (props) => {
  const { textStyle, viewStyle, smallerTextStyle } = styles;
  return (
    <View style={viewStyle}>
    <Text style={textStyle}>SunInformed</Text>
    <Text style={smallerTextStyle}>{props.headerText}</Text>
    </View>
  );
};

const styles = {
  textStyle: {
    fontSize: 14
  },
  smallerTextStyle: {
    fontSize: 9,
    marginTop: 5
  },
  viewStyle: {
    backgroundColor: '#F8F8F8',
    justifyContent: 'center',
    alignItems: 'center',
    height: 65,
    paddingTop: 15,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.5
  }
};

// export the component making it available to other parts of the app
export default Header;
