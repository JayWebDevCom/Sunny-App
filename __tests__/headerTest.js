import 'react-native';
import React from 'react';
import Header from '../src/components/Header';
import renderer from 'react-test-renderer';

test('sunny app renders header correctly', () => {
  const tree = renderer.create(
    <Header />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
