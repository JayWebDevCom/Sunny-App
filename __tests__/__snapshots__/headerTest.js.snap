// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sunny app renders header correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F8F8F8",
      "height": 65,
      "justifyContent": "center",
      "paddingTop": 15,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.5,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 14,
      }
    }
  >
    SunInformed
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 9,
        "marginTop": 5,
      }
    }
  />
</View>
`;
